WordPress/Classes/System/WordPressAppDelegate.m

```diff
@@ -204,29 +202,19 @@ - (void)setupLookback
 #ifdef LOOKBACK_ENABLED
     // Kick this off on a background thread so as to not slow down the app initialization
     dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_BACKGROUND, 0), ^{
-        if ([WordPressComApiCredentials lookbackToken].length > 0) {
-            [Lookback setupWithAppToken:[WordPressComApiCredentials lookbackToken]];
-            [[NSUserDefaults standardUserDefaults] registerDefaults:@{WPInternalBetaShakeToPullUpFeedbackKey: @YES}];
-            [[NSUserDefaults standardUserDefaults] setObject:@(NO) forKey:LookbackCameraEnabledSettingsKey];
-            [Lookback lookback].shakeToRecord = [[NSUserDefaults standardUserDefaults] boolForKey:WPInternalBetaShakeToPullUpFeedbackKey];
-            
-            // Setup Lookback to fire when the user holds down with three fingers for around 3 seconds
-            dispatch_async(dispatch_get_main_queue(), ^{
-                UILongPressGestureRecognizer *recognizer = [[UILongPressGestureRecognizer alloc] initWithTarget:self action:@selector(lookbackGestureRecognized:)];
-                recognizer.minimumPressDuration = 3;
-                recognizer.cancelsTouchesInView = NO;
-#if TARGET_IPHONE_SIMULATOR
-                recognizer.numberOfTouchesRequired = 2;
-#else
-                recognizer.numberOfTouchesRequired = 3;
-#endif
-                [[UIApplication sharedApplication].keyWindow addGestureRecognizer:recognizer];
-            });
-            
+        
+        NSString *lookbackToken = [WordPressComApiCredentials lookbackToken];
+        
+        if ([lookbackToken length] > 0) {
+            UIWindow *keyWindow = [UIApplication sharedApplication].keyWindow;
+
             NSManagedObjectContext *context = [[ContextManager sharedInstance] mainContext];
             AccountService *accountService = [[AccountService alloc] initWithManagedObjectContext:context];
             WPAccount *account = [accountService defaultWordPressComAccount];
```

Comment:
I know you didn't actually change this code, but i think it's a potential source of issues. The `mainContext` should be always used while running on the main thread.

But in this particular scenario, we're running in a background queue. This should probably be wrapped as follows:

```
[context performBlock:^{
            AccountService *accountService = [[AccountService alloc] initWithManagedObjectContext:context];
             WPAccount *account = [accountService defaultWordPressComAccount];
             self.lookbackPresenter = [[WPLookbackPresenter alloc] initWithToken:lookbackToken
                                                                         userId:account.username
                                                                         window:keyWindow];
}];
```